qiskit_ibm_runtime/sampler.py

```diff
@@ -265,23 +264,10 @@ def _call(
             options=Options._get_runtime_options(combined),
             result_decoder=SamplerResultDecoder,
         ).result()
-        quasi_dists = []
-        for quasi, meta in zip(raw_result["quasi_dists"], raw_result["metadata"]):
-            shots = meta.get("shots", float("inf"))
-            overhead = meta.get("readout_mitigation_overhead", 1.0)
-
-            # M3 mitigation overhead is gamma^2
-            # https://github.com/Qiskit-Partners/mthree/blob/423d7e83a12491c59c9f58af46b75891bc622949/mthree/mitigation.py#L457
-            #
-            # QuasiDistribution stddev_upper_bound is gamma / sqrt(shots)
-            # https://github.com/Qiskit/qiskit-terra/blob/ff267b5de8b83aef86e2c9ac6c7f918f58500505/qiskit/result/mitigation/local_readout_mitigator.py#L288
-            stddev = sqrt(overhead / shots)
-            quasi_dists.append(
-                QuasiDistribution(quasi, shots=shots, stddev_upper_bound=stddev)
-            )
+        quasi_dists = convert_quasi_dists(raw_result)
```

Comment:
I don't think this would actually work. Given the call to `run()` above uses `SamplerResultDecoder`, which would already convert the result to `QuasiDistribution`. Maybe you can wait for #524 first.